packages/generic-utils/src/create-deferred/create-deferred.ts

```diff
@@ -4,13 +4,22 @@
 export const createDeferred = <T>() => {
   let resolve!: (value: T | PromiseLike<T>) => void
   let reject!: (reason: any) => void
+  let settled = false
   const promise = new Promise<T>((_resolve, _reject) => {
-    resolve = _resolve
-    reject = _reject
+    resolve = (...args) => {
+      settled = true
+      _resolve(...args)
+    }
+    reject = (...args) => {
+      settled = true
+      _reject(...args)
+    }
```

Comment:
Ah, nice approach!

Reply:
Can't believe it didn't come to mind before